src/components/menu/__internal__/submenu/submenu.component.tsx

```diff
@@ -297,40 +296,15 @@ const Submenu = React.forwardRef<
           let nextIndex = index;
 
           if (href && !submenuFocusId) {
-            if (
-              Events.isDownKey(event) ||
-              Events.isUpKey(event) ||
-              (Events.isTabKey(event) && !Events.isShiftKey(event))
-            ) {
+            if (Events.isDownKey(event) || Events.isUpKey(event)) {
```

Comment:
question: are these changes within the scope of this commit (i.e. upgrading React to v18) as well?